File: PATTERNS/Pattern48/Pattern48.c
#include <stdio.h>

void pattern48(int n);

int main()
{
    int n = 0;
    printf("Enter the number of rows: ");
    scanf("%d", &n);
    printf("\n");
    pattern48(n);
}

void pattern48(int n)
{
    int count = 0;

    for (int i = 1; i <= n; i++){
        if (i % 2 == 0){
            count += i;
            for (int j = i - 1; j >= 0; j--){
                printf("%d ", count);
                count--;
            }
            count += i;
        }
        else{
            for (int j = 0; j < i; j++){
                count++;
                printf("%d ", count);
            }
        }
         
        printf("\n");
    }
    printf("\n");
}

File: PATTERNS/Pattern48/Pattern48.go
package main

import "fmt"

func main() {
	var n,I int
	fmt.Print("Enter the row number:")
	fmt.Scanf("%d",&n)
	
	for k,i := 1,1;i<=n;i++{
		I = k+i-1;
		for j:=1;j<=i;j++{
			if i%2==1{
				fmt.Printf("%d ",k);
			}else{
				fmt.Printf("%d ",I);
			}
			k++;
			I--;
		}
		fmt.Println()
	}
}